### PR TITLE
PIM-4292: Handle asset categories permissions

### PR DESCRIPTION
**Pull request by skeleton**
| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | Y 
| Specs & Static     | Y
| Changelog updated  | Y

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # 1.4.x
 
 ## BC breaks
+- Change the constructor of `Pim\Bundle\UserBundle\Context\UserContext`. Takes `Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface`, `Pim\Bundle\CatalogBundle\Repository\LocaleRepositoryInterface`, `Pim\Bundle\CatalogBundle\Repository\ChannelRepositoryInterface`, `Pim\Component\Classification\Repository\CategoryRepositoryInterface`, `Symfony\Component\HttpFoundation\RequestStack`, `Pim\Bundle\CatalogBundle\Builder\ChoicesBuilderInterface` and a `$defaultLocale` string
```
Comment (jjanvier) [inline CHANGELOG.md]: will go in CHANGELOG 1.4.md

File: src/Pim/Bundle/EnrichBundle/DependencyInjection/PimEnrichExtension.php [hunk 2]
```diff
@@ -32,11 +32,13 @@ public function load(array $configs, ContainerBuilder $container)
 
         $loader = new YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('attribute_icons.yml');
+        $loader->load('category_counters.yml');
         $loader->load('colors.yml');
         $loader->load('command.yml');
         $loader->load('connector/cleaners.yml');
         $loader->load('connector/processors.yml');
         $loader->load('connector/readers.yml');
+        $loader->load('category_counters.yml');
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/DependencyInjection/PimEnrichExtension.php]: twice

**Comment by jjanvier** [comment]
GTM

File: src/Pim/Bundle/EnrichBundle/Controller/CategoryTreeController.php [hunk 3]
```diff
@@ -60,94 +52,80 @@ class CategoryTreeController extends AbstractDoctrineController
...
      */
     public function listTreeAction(Request $request)
     {
-        $selectNodeId = $request->get('select_node_id', -1);
-        $relatedEntity = $this->getRequest()->get('related_entity', 'product');
+        if (false === $this->securityFacade->isGranted($this->buildAclName('category_list'))) {
+            throw new AccessDeniedException();
+        }
+
+        $selectNodeId  = $request->get('select_node_id', -1);
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/CategoryTreeController.php]: extra space

File: src/Pim/Bundle/EnrichBundle/Controller/CategoryTreeController.php [hunk 4]
```diff
@@ -182,24 +172,25 @@ public function moveNodeAction(Request $request)
...
         } catch (NotFoundHttpException $e) {
             return ['categories' => []];
         }
 
-        $selectNodeId   = $this->getRequest()->get('select_node_id', -1);
-        $withItemsCount = (bool)$this->getRequest()->get('with_items_count', false);
-        $includeParent  = (bool)$this->getRequest()->get('include_parent', false);
-        $includeSub     = (bool)$this->getRequest()->get('include_sub', false);
-        $relatedEntity  = $this->getRequest()->get('related_entity', 'product');
+        $selectNodeId   = $request->get('select_node_id', -1);
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/CategoryTreeController.php]: extra space

File: src/Pim/Bundle/EnrichBundle/Controller/CategoryTreeController.php [hunk 5]
```diff
@@ -327,20 +347,26 @@ public function editAction(Request $request, $id)
...
     {
+        if (false === $this->securityFacade->isGranted($this->buildAclName('category_remove'))) {
+            throw new AccessDeniedException();
+        }
+
         $category = $this->findCategory($id);
-        $parent = $category->getParent();
-        $params = ($parent !== null) ? ['node' => $parent->getId()] : [];
+        $parent   = $category->getParent();
+        $params   = ($parent !== null) ? ['node' => $parent->getId()] : [];
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/CategoryTreeController.php]: yoda

File: src/Pim/Bundle/EnrichBundle/Resources/views/Product/index.html.twig [hunk 6]
```diff
@@ -50,7 +50,7 @@
     {% set categoriesGranted = resource_granted('pim_enrich_product_category_list') %}
     <div class="row-fluid {{ categoriesGranted ? 'has-sidebar': '' }}">
         {% if categoriesGranted %}
-            <div id="tree" data-datalocale="{{ dataLocale.code }}" data-relatedentity="product"></div>
+            <div id="tree" data-datalocale="{{ dataLocale.code }}"></div>
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/views/Product/index.html.twig]: :+1:

File: src/Pim/Bundle/EnrichBundle/Twig/CategoryExtension.php [hunk 7]
```diff
@@ -173,23 +166,27 @@ public function listCategoriesResponse(
      *
      * @param CategoryInterface $category
      * @param bool              $includeSub
+     * @param string            $relatedEntity
      *
      * @return bool
+     * @throws \Exception
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Twig/CategoryExtension.php]: wrong order and missing blank line

**Pull request merged**